### ISSUE TITLE
[AVM Module Issue]: diff_disk_settings not working for linux VM

### ISSUE THREAD

**Issue by Laffs2k5** [Azure/terraform-azurerm-avm-res-compute-virtualmachine#37]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Module specific issue

- [X] I confirm that this issue is about this AVM module and not about the resource itself, or how to use Terraform.

### Issue Type?

Bug

### (Optional) Module Version

0.7.0

### Description

The `diff_disk_settings` of input variable `os_disk` does not work for Linux VM's. 

This code: https://github.com/Azure/terraform-azurerm-avm-res-compute-virtualmachine/blob/v0.7.0/main.linux_vm.tf#L53-L60
```hcl
    dynamic "diff_disk_settings" {
      for_each = var.os_disk.diff_disk_settings == null ? [] : ["diff_disk_settings"]

      content {
        option    = var.diff_disk_settings.option
        placement = var.diff_disk_settings.placement
      }
    }
```

should be (notice the difference inside the `content` block):
```hcl
    dynamic "diff_disk_settings" {
      for_each = var.os_disk.diff_disk_settings == null ? [] : ["diff_disk_settings"]

      content {
        option    = var.os_disk.diff_disk_settings.option
        placement = var.os_disk.diff_disk_settings.placement
      }
    }
```
Copied from `main.windows_vm.tf` which seems to be correctly declared.

**Comment by jchancellor-ms** [comment]
@Laffs2k5  - Thanks for catching this bug.  I'll add the fix and push a release.

**Comment by jchancellor-ms** [comment]
@Laffs2k5 - this has been corrected in version 0.8.0.  I'm closing this but can re-open if this doesn't correct your issue.
